scripts/uberenv/uberenv.py

```diff
@@ -188,127 +195,445 @@ def load_json_file(json_file):
-    """
-    Returns the path of a site scoped spack mirror with the
-    given name, or None if no mirror exists.
-    """
-    rv, res = sexe("spack/bin/spack mirror list", ret_output=True)
-    mirror_path = None
-    for mirror in res.split('\n'):
-        if mirror:
-            parts = mirror.split()
-            if parts[0] == mirror_name:
-                mirror_path = parts[1]
-    return mirror_path
-
-
-def use_spack_mirror(spack_dir,
-                     mirror_name,
-                     mirror_path):
-    """
-    Configures spack to use mirror at a given path.
-    """
-    mirror_path = os.path.abspath(mirror_path)
-    existing_mirror_path = find_spack_mirror(spack_dir, mirror_name)
-    if existing_mirror_path and mirror_path != existing_mirror_path:
-        # Existing mirror has different URL, error out
-        print "[removing existing spack mirror `%s` @ %s]" % (mirror_name,
-                                                              existing_mirror_path)
-        #
-        # Note: In this case, spack says it removes the mirror, but we still
-        # get errors when we try to add a new one, sounds like a bug
-        #
-        sexe("spack/bin/spack mirror remove --scope=site {} ".format(
-                mirror_name), echo=True)
-        existing_mirror_path = None
-    if not existing_mirror_path:
-        # Add if not already there
-        sexe("spack/bin/spack mirror add --scope=site {} {}".format(
-                mirror_name, mirror_path), echo=True)
-        print "[using mirror %s]" % mirror_path
+        # setup main package name
+        if opts["install"]:
+            self.pkg_name = self.project_opts["package_name"]
+        else:
+            self.pkg_name = self.project_opts["uberenv_package_name"]
+
+    def setup_paths_and_dirs(self):
+        self.uberenv_path = os.path.split(os.path.abspath(__file__))[0]
+
+        self.dest_dir = os.path.abspath(self.opts["prefix"])
+        print("[installing to: {0}]".format(self.dest_dir))
+
+        # print a warning if the dest path already exists
+        if not os.path.isdir(self.dest_dir):
+            os.mkdir(self.dest_dir)
+        else:
+            print("[info: destination '{}' already exists]".format(self.dest_dir))
+
+    def detect_platform(self):
+        # find supported sets of compilers.yaml, packages,yaml
+        res = None
+        if is_darwin():
+            res = "darwin"
+        elif "SYS_TYPE" in os.environ.keys():
+            sys_type = os.environ["SYS_TYPE"].lower()
+            res = sys_type
+        return res
+
+
+class VcpkgEnv(UberEnv):
+    """ Helper to clone vcpkg and install libraries on Windows """
+
+    def __init__(self, opts, extra_opts):
+        UberEnv.__init__(self,opts,extra_opts)
+
+        # setup architecture triplet
+        self.triplet = opts["triplet"]
+        if self.triplet is None:
+           self.triplet = os.getenv("VCPKG_DEFAULT_TRIPLET", "x86-windows")
```

Comment:
Perhaps so.  We should get the triplet from (in descending order of precedence):
1. user's command line
2. user's environment variable `VCPKG_DEFAULT_TRIPLET`, which is where vcpkg looks
3. `projects.json`
4. lacking any of 1--3, we need to use `x86-windows` since that's vcpkg's final default

Reply:
Sounds good to me.